### PR TITLE
mtl/ofi: use libfabric 1.18 API if available

### PR DESCRIPTION
**Pull request by wzamazon**
This patch try to use 1.18 API when it is available. This is because 1.18 API clearly define provider's CUDA API behavior to be that provider can call CUDA API by default if application uses 1.18 API.

When using older version of API, some libfabric will not claim support of FI_HMEM even if it is capable of supporting because the provider does not know whether CUDA calls are permitted.

### CODE REVIEW

**Comment by wzamazon** [comment]
bot:aws:retest

**Comment by wckzhang** [comment]
Can you also make changes in the btl/ofi?

File: ompi/mca/mtl/ofi/mtl_ofi_component.c [hunk 1]
```diff
@@ -695,7 +704,11 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
         hints_dup->caps &= ~(FI_LOCAL_COMM | FI_REMOTE_COMM);
         hints_dup->fabric_attr->prov_name = strdup("efa");
 
-        ret = fi_getinfo(fi_version, NULL, NULL, 0ULL, hints_dup, &providers);
+        ret = fi_getinfo(fi_primary_version, NULL, NULL, 0ULL, hints_dup, &providers);
+        if (FI_ENOSYS == -ret) {
```
Comment (wckzhang) [inline ompi/mca/mtl/ofi/mtl_ofi_component.c:708]: I couldn't find in the man page that FI_ENOSYS is returned when version doesn't match, if this is the case, is it possible to add it to the man page?

Comment (wzamazon) [reply ompi/mca/mtl/ofi/mtl_ofi_component.c:708]: Sure. I am engaging with the community to try to clarify

Comment (wzamazon) [reply ompi/mca/mtl/ofi/mtl_ofi_component.c:708]: Libfabric document updated:

https://ofiwg.github.io/libfabric/main/man/fi_getinfo.3.html

**Review by wckzhang — APPROVED** [review]

**Review by hppritcha — APPROVED** [review]

**Comment by wzamazon** [comment]
@wckzhang  @hppritcha

Can you help merge this PR?

**Comment by hppritcha** [comment]
@wzamazon please cherry pick back to 5.0.x

**Pull request merged**